src/lib/y2storage/planned/can_be_formatted.rb

```diff
@@ -115,15 +116,32 @@ def reformat?
         reformat
       end
 
+      # Whether activation and configurion of Btrfs snapshots is wanted, if
+      # possible, in the resulting filesystem.
+      #
+      # @see #snapshots?
+      attr_writer :snapshots
+
+      # Whether Btrfs snapshots should be activated in the resulting filesystem.
+      #
+      # @return [Boolean] true if snapshots are requested (see {#snapshots=}) and
+      #   possible (so far, only for Btrfs root filesystems).
+      def snapshots?
+        @snapshots && btrfs? && mount_point == "/"
+      end
+
     protected
 
       # Creates subvolumes in the previously created filesystem that is placed
       # in the final device.
       #
-      # This also sets other Btrfs attributes, like the default subvolume.
+      # This also sets other Btrfs attributes, like the default subvolume or
+      # Filesystems::Btrfs#configure_snapper
       #
       # @param filesystem [Filesystems::BlkFilesystem]
       def btrfs_setup(filesystem)
+        filesystem.configure_snapper = snapshots? if filesystem.respond_to?(:configure_snapper=)
```

Comment:
Does it have sense to "configure snapper" if the fs does not support btrfs subvolumes? Maybe to move this line after `return unless filesystem.supports_btrfs_subvolumes?` ?

Reply:
In theory, Snapper can work with other filesystem types, although so far we don't consider it. So supporting snapper and supporting btrfs subvolumes are different questions.